src/ripple/nodestore/impl/DatabaseShardImp.cpp

```diff
@@ -830,131 +865,172 @@ DatabaseShardImp::import(Database& source)
                 continue;
             }
 
-            // Verify SQLite ledgers are in the node store
-            {
-                auto const firstSeq{firstLedgerSeq(shardIndex)};
-                auto const lastSeq{
-                    std::max(firstSeq, lastLedgerSeq(shardIndex))};
-                auto const numLedgers{
-                    shardIndex == earliestShardIndex() ? lastSeq - firstSeq + 1
-                                                       : ledgersPerShard_};
-                auto ledgerHashes{getHashesByIndex(firstSeq, lastSeq, app_)};
-                if (ledgerHashes.size() != numLedgers)
-                    continue;
-
-                bool valid{true};
-                for (std::uint32_t n = firstSeq; n <= lastSeq; n += 256)
-                {
-                    if (!source.fetchNodeObject(ledgerHashes[n].first, n))
-                    {
-                        JLOG(j_.warn()) << "SQLite ledger sequence " << n
-                                        << " mismatches node store";
-                        valid = false;
-                        break;
-                    }
-                }
-                if (!valid)
-                    continue;
-            }
+            importStatus_->currentIndex = shardIndex;
+        }
 
-            auto const path =
-                needsHistoricalPath ? chooseHistoricalPath(lock) : dir_;
+        std::uint32_t const firstSeq = firstLedgerSeq(shardIndex);
+        std::uint32_t const lastSeq =
+            std::max(firstSeq, lastLedgerSeq(shardIndex));
+        std::uint32_t const numLedgers = shardIndex == earliestShardIndex()
+            ? lastSeq - firstSeq + 1
+            : ledgersPerShard();
 
-            // Create the new shard
-            auto shard{
-                std::make_unique<Shard>(app_, *this, shardIndex, path, j_)};
-            if (!shard->init(scheduler_, *ctx_))
+        // Verify SQLite ledgers are in the node store
+        {
+            auto const ledgerHashes{getHashesByIndex(firstSeq, lastSeq, app_)};
+            if (ledgerHashes.size() != numLedgers)
                 continue;
 
-            // Create a marker file to signify an import in progress
-            auto const shardDir{path / std::to_string(shardIndex)};
-            auto const markerFile{shardDir / importMarker_};
+            auto constexpr stepSize = 256;
```

Comment:
This predates me but it's done this way to give a rough early indication of whether or not we've got the necessary ledgers. Will change so that every ledger is checked.

Reply:
We should be checking every ledger (step 1). Prior was every 256 as a way to quickly give tell us if we believed enough data was present for a complete shard. That was okay with full history cases but with this PR we should be checking if every ledger is present.